platforms/nuttx/src/px4/common/tasks.cpp

```diff
@@ -92,7 +92,7 @@ int px4_task_delete(int pid)
 
 	if (pid == getpid()) {
 		// Commit suicide
-		exit(EXIT_SUCCESS);
```

Comment:
This needs to be flagged with __KERNEL__. I think user space should still call exit() ?